crates/sncast/src/response/structs.rs

```diff
@@ -122,3 +122,57 @@ pub struct ScriptInitResponse {
 }
 
 impl CommandResponse for ScriptInitResponse {}
+
+#[derive(Serialize)]
+pub enum FinalityStatus {
+    Received,
+    Rejected,
+    AcceptedOnL2,
+    AcceptedOnL1,
+}
+
+#[derive(Serialize)]
+pub enum ExecutionStatus {
+    Succeeded,
+    Reverted,
+}
+
+#[derive(Serialize)]
+pub struct TransactionStatusResponse {
+    pub finality_status: FinalityStatus,
+    pub execution_status: Option<ExecutionStatus>,
+}
+
+impl From<TransactionStatus> for TransactionStatusResponse {
+    fn from(status: TransactionStatus) -> Self {
+        match status {
+            TransactionStatus::Received => Self {
+                finality_status: FinalityStatus::Received,
+                execution_status: None,
+            },
+            TransactionStatus::Rejected => Self {
+                finality_status: FinalityStatus::Rejected,
+                execution_status: None,
+            },
+            TransactionStatus::AcceptedOnL2(execution_status) => Self {
+                finality_status: FinalityStatus::AcceptedOnL2,
+                execution_status: Some(execution_status.into()),
+            },
+            TransactionStatus::AcceptedOnL1(execution_status) => Self {
+                finality_status: FinalityStatus::AcceptedOnL1,
+                execution_status: Some(execution_status.into()),
+            },
+        }
+    }
+}
+
+impl From<TransactionExecutionStatus> for ExecutionStatus {
+    fn from(status: TransactionExecutionStatus) -> Self {
+        match status {
+            TransactionExecutionStatus::Succeeded => Self::Succeeded,
+            TransactionExecutionStatus::Reverted => Self::Reverted,
+        }
+    }
+}
```

Comment:
Imo those should not be implemented here, but in `tx_status.rs` instead. This file defines output structs, it shouldn't contain information about the sources of those.

Reply:
I've also considered that, but `starknet_commands` is a separate crate and due to the orphan rule, we can't implement it there. What about defining `impl.rs` in the `response` directory?

Reply:
Good point, I've missed that. Imo we should just use functions instead of traits. It is better than mixing the abstraction layers.